ntoskrnl/ke/i386/mp.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture source file to hold multiprocessor functions
+ * PROGRAMMERS:  Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
+
+struct APInfo
+{
+    KPCR pcr;
+    KTSS tss;
+    ETHREAD thread;
+    KDESCRIPTOR GdtDesc;
+    KDESCRIPTOR IdtDesc;
+} APInfo;
+
+/* FUNCTIONS *****************************************************************/
+VOID
+NTAPI
+KeStartAllProcessors()
+{
+    BOOLEAN ApplicationProcessorsEnabled = FALSE;
+    HighestPhysicalAddress.QuadPart = UINT64_MAX;
+    KDESCRIPTOR GdtDesc, IdtDesc;
+    __sgdt(&GdtDesc.Limit);
+    __sidt(&IdtDesc.Limit);
+    do
+    {
+        ProcessorCount++;
+
+        /* Attempt to allocate memory for its new setup */
+        SIZE_T APInfoSize = sizeof(struct APInfo) + 2;
+        PVOID PAPInfo = MmAllocateContiguousMemory(APInfoSize, HighestPhysicalAddress);
+
+        if (!PAPInfo)
+        {
+            ASSERT("KeStartAllProcessors: Memory Allocation has failed");
+        }
+
+        RtlCopyMemory((PVOID)((ULONG_PTR)PAPInfo + sizeof(KPCR) + sizeof(KTSS)), (PVOID)GdtDesc.Base, GdtDesc.Limit + 1);
```

Comment:
Treat `PAPInfo` like a pointer to a structure, you won't need `RtlCopyMemory` that way

Reply:
You're writing into the ETHREAD part of PAPInfo.

Reply:
Yeah, and I don't think the entire GDT fits into the KDESCRIPTOR field either.